pkg/util/error_list.go

```diff
@@ -27,10 +27,10 @@ import (
 
 // This helper implements the error interface for []error, but prevents
 // accidental conversion of []error to error.
-type errorList []error
+type ErrorList []error
```

Comment:
Having read over just this portion of the PR< I think we want a different strategy.

What about something like:

1) break the existing SliceToError() stuff into pkg/util/errlist (and rename to errlist.ToError(errs)).
2) add your logic as errlist.FilterOut(errs []error, funcs ...MatchFunc) there

Then we can argue about flattening...